From 673b9ba7a1404fa5beda936b8ad509b70a516b52 Mon Sep 17 00:00:00 2001
From: Yang Yingliang <yangyingliang@huawei.com>
Date: Sat, 29 May 2021 16:20:57 +0800
Subject: [PATCH] cifsd: fix memleak in ksmbd_vfs_stream_read()
Git-commit: 673b9ba7a1404fa5beda936b8ad509b70a516b52
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Before ksmbd_vfs_stream_read() return, memory allocate in
ksmbd_vfs_getcasexattr() need be freed.

Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/vfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/cifsd/vfs.c b/fs/cifsd/vfs.c
index e70b67e41cd4..85872416bf9b 100644
--- a/fs/cifsd/vfs.c
+++ b/fs/cifsd/vfs.c
@@ -290,6 +290,7 @@ static int ksmbd_vfs_stream_read(struct ksmbd_file *fp, char *buf, loff_t *pos,
 	}
 
 	memcpy(buf, &stream_buf[*pos], count);
+	kvfree(stream_buf);
 	return v_len > count ? count : v_len;
 }
 
-- 
2.33.0

